internal/images/custom-scorecard-tests/main.go

```diff
@@ -0,0 +1,134 @@
+// either these or other terms.
+//
+// This license is subject to the following condition:
+// The above copyright notice and either this complete permission notice or at
+// a minimum a reference to the UPL must be included in all copies or
+// substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	scapiv1alpha3 "github.com/operator-framework/api/pkg/apis/scorecard/v1alpha3"
+	apimanifests "github.com/operator-framework/api/pkg/manifests"
+
+	tests "github.com/cryostatio/cryostat-operator/internal/test/scorecard"
+)
+
+const podBundleRoot = "/bundle"
+
+func main() {
+	entrypoint := os.Args[1:]
+	if len(entrypoint) == 0 {
+		log.Fatal("specify one or more test name arguments")
+	}
+
+	// Get namespace from SCORECARD_NAMESPACE environment variable
+	namespace := os.Getenv("SCORECARD_NAMESPACE")
+	if len(namespace) == 0 {
+		log.Fatal("SCORECARD_NAMESPACE environment variable not set")
+	}
+
+	// Read the pod's untar'd bundle from a well-known path.
+	bundle, err := apimanifests.GetBundleFromDir(podBundleRoot)
+	if err != nil {
+		log.Fatalf("failed to read bundle manifest: %s", err.Error())
+	}
+
+	var results []scapiv1alpha3.TestResult
+
+	// Check that test arguments are valid
+	if !validateTests(entrypoint) {
+		results = printValidTests()
+	} else {
+		results = runTests(entrypoint, bundle, namespace)
+	}
+
+	// Print results in expected JSON form
+	printJSONResults(results)
+}
+
+func printValidTests() []scapiv1alpha3.TestResult {
+	result := scapiv1alpha3.TestResult{}
+	result.State = scapiv1alpha3.FailState
+	result.Errors = make([]string, 0)
+	result.Suggestions = make([]string, 0)
+
+	str := fmt.Sprintf("valid tests for this image include: %s", strings.Join([]string{
+		tests.OperatorInstallTestName,
+	}, ","))
+	result.Errors = append(result.Errors, str)
+
+	return []scapiv1alpha3.TestResult{result}
+}
+
+func validateTests(testNames []string) bool {
+	for _, testName := range testNames {
+		switch testName {
+		case tests.OperatorInstallTestName:
+		default:
+			return false
+		}
+	}
+	return true
+}
+
+func runTests(testNames []string, bundle *apimanifests.Bundle, namespace string) []scapiv1alpha3.TestResult {
+	results := []scapiv1alpha3.TestResult{}
+
+	// Run tests
+	for _, testName := range testNames {
+		switch testName {
+		case tests.OperatorInstallTestName:
+			results = append(results, tests.OperatorInstallTest(bundle, namespace))
+		default:
```

Comment:
Looks great!  I just noticed since we validateTests, this default case won't be reachable right?

Reply:
Not normally, it's just there as a sanity test if we add a new test to `validateTests` but forget to add it to `runTests`.

Reply:
Right! That is a good idea :D